### ISSUE TITLE
Move all source code into an `src` root folder

### ISSUE THREAD

**Issue by myssto** [osu-tournament-rating/otr-web#225]
As of now, all source code for the app lives in the root of the repository alongside all configs and repo files. We should move this all into `/src` for better organization. Refactoring should ensure that all module imports are updated, CI / CD actions are checked for paths, etc.

Current:
```
/
|-- app/
    |-- *app router files*
|-- components/
    |-- *component files*
|-- middleware.ts
|-- package.json
|-- tsconfig.json
|-- eslint.rc
...
```

Proposed:
```
/
|-- src/
    |-- app/
        |-- *app router files*
    |-- components/
        |-- *component files*
    |-- middleware.ts
|-- package.json
|-- tsconfig.json
|-- eslint.rc
...
```

**Comment by hburn7** [comment]
Did we want to do this in the new site? Personally I don't think it's very necessary and by default shadcn generates projects without the src folder.

**Comment by myssto** [comment]
> Did we want to do this in the new site? Personally I don't think it's very necessary and by default shadcn generates projects without the src folder.

Im not married to it. Just so we're on the same page though, shadcn has nothing to do with this at all. It's a component library for react, it doesn't (and shouldn't) control any convention imo. And fwiw the npx script you used to setup the project uses `create-next-app` under the hood, which supports an src directory out of the box. Using an src directory for projects is one of the oldest javascript conventions you can find :P

**Comment by hburn7** [comment]
Yeah if we use a src folder we will need to edit some configs so shadcn CLI knows where to put things, but if we're okay with that then we can use the src folder.
